core/src/main/scala/cats/Foldable.scala

```diff
@@ -152,6 +152,26 @@ import simulacrum.typeclass
   def size[A](fa: F[A]): Long = foldMap(fa)(_ => 1)
 
   /**
+    * Get the element at the index of the `Foldable`.
+    *
+    * The default implementation of this is based on `foldLeft`, and thus will
+    * always fold across the entire structure.
+    */
+  def get[A](fa: F[A])(idx: Long): Option[A] =
+    foldLeft(fa, (0L, Option.empty[A])) { (b, a) =>
+      b._1 match {
+        case i if i == idx  => (i + 1, Some(a))
+        case i if i < idx => (i + 1, None)
+        case _ => b
+      }
+    }._2
+
+  /**
+    * Get the element at the index of the `Foldable`
+    */
+  def apply[A](fa: F[A])(idx: Long): A = get(fa)(idx).get
```

Comment:
@jedws I disagree with the assertion that only completely safe FP functions should exist in cats. If that were the case, then no unlawful instances (see `Seq`, `Try`) should exist in cats at all, and  cats ought to not dependent on the stdlib for _any_ of its collections because the functions on them such as `map` and `flatMap` are already a mess.

I also don't think it's necessarily bad behaviour to call `apply`; in the case that you know for a fact that a key exists at compile time, why should a user not call `get.get`? Is there any real point in adding extra barriers when the user is clearly aware that it's unsafe and calling it anyway?